### PR TITLE
[Merged by Bors] - feat: if colimits of commute with finite limits, then category is filtered

### PR DESCRIPTION
**Pull request by TwoFX**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Filtered/OfColimitCommutesFiniteLimit.lean [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+theorem isFiltered_of_nonempty_limit_colimit_to_colimit_limit
+    (h : ∀ {J : Type v} [SmallCategory J] [FinCategory J] (F : J ⥤ K ⥤ Type v),
+      Nonempty (limit (colimit F.flip) ⟶ colimit (limit F))) : IsFiltered K := by
+  refine IsFiltered.iff_nonempty_limit.2 (fun {J} _ _ F => ?_)
+  suffices Nonempty (limit (colimit (F.op ⋙ coyoneda).flip)) by
+    obtain ⟨X, y, -⟩ := Types.jointly_surjective' (this.map (h (F.op ⋙ coyoneda)).some).some
+    exact ⟨X, ⟨(limitObjIsoLimitCompEvaluation (F.op ⋙ coyoneda) _).hom y⟩⟩
+  let _ (j : Jᵒᵖ) : Unique ((colimit (F.op ⋙ coyoneda).flip).obj j) :=
+    ((colimitObjIsoColimitCompEvaluation (F.op ⋙ coyoneda).flip _
+      ≪≫ Coyoneda.colimitCoyonedaIso _)).toEquiv.unique
```
Comment (joelriou) [inline Mathlib/CategoryTheory/Filtered/OfColimitCommutesFiniteLimit.lean:32]:
```suggestion
    ((colimitObjIsoColimitCompEvaluation (F.op ⋙ coyoneda).flip _ ≪≫
      Coyoneda.colimitCoyonedaIso _)).toEquiv.unique
```

**Comment by joelriou** [comment]
Thanks!

bors d+

File: Mathlib/CategoryTheory/Filtered/OfColimitCommutesFiniteLimit.lean (hunk 1 above)
Comment (TwoFX) [reply Mathlib/CategoryTheory/Filtered/OfColimitCommutesFiniteLimit.lean]: Thanks, I'm not sure why this particular style rule is so hard for me to remember...

**Pull request closed without merging**